addons/account_payment/models/payment_provider.py

```diff
@@ -98,11 +98,19 @@ def _check_existing_payment_method_lines(self, payment_method):
                 payment_method.id)], limit=1)
         return bool(existing_payment_method_lines_count)
 
+    def _check_existing_payment(self, payment_method):
+        existing_payment_count = self.env['account.payment'].search_count([('payment_method_id', '=', payment_method.id)], limit=1)
+        return bool(existing_payment_count)
+
     @api.model
     def _remove_provider(self, code):
         """ Override of `payment` to delete the payment method of the provider. """
         payment_method = self._get_provider_payment_method(code)
-        if self._check_existing_payment_method_lines(payment_method):
-            raise UserError(_("You cannot uninstall this module as payment method transactions already exist."))
+        # If the payment method is used by any payments, we block the uninstallation of the module.
+        if self._check_existing_payment(payment_method):
+            raise UserError(_("You cannot uninstall this module as payments using this payment method already exist."))
         super()._remove_provider(code)
+        # If it is not used by a payment but is used by a payment method line, we delete the payment method line.
+        if self._check_existing_payment_method_lines(payment_method):
+            self.env['account.payment.method.line'].search([('payment_method_id', '=', payment_method.id)]).unlink()
```

Comment:
Why do we need to do that? It seems there is a `ondelete='cascade'` set on the field :thinking: 
https://github.com/odoo/odoo/blob/6021fd883f78f41dea6c2a07fb970e230225b995/addons/account/models/account_payment_method.py#L99-L105